### DISCUSSION TITLE
lesson 9

### DISCUSSION THREAD

**Discussion by cell0901** [Cyfrin/foundry-full-course-cu#814] [Q&A]
why are we using the **vm.deal**  in the **setUp()** function and how does every test function knows it? why don't we assign it in every  test function where we need to work with funds

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/142923689/b7c38458-604b-44a7-9ec2-4b2adcf92694)

**Comment by shikhar229169** [comment]
deal is a cheatcode in foundry which is used to provide fake eth to the user. As we perform a function call by the user, it costs some gas that's why we need to provide some initial funds to the user.
Refer to the docs: https://book.getfoundry.sh/cheatcodes/deal#deal

Just before running a test function, everytime setUp function is called, thus we write it in setUp as it is common for every test.

**Reply by cell0901** [reply]
so forge calls setUp() function first? is it something predefined in the library?
